cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -552,6 +552,23 @@ impl From<sp_blockchain::Error> for GossipMessageError {
 	}
 }
 
+/// Compares if the receipt `other` is the same with `local`, return a tuple of (local_index,
+/// local_trace_root) if there is a mismatch.
+pub(crate) fn find_trace_mismatch<Number, Hash: Copy + Eq, PNumber, PHash>(
+	local: &ExecutionReceipt<Number, PHash, Hash>,
+	other: &ExecutionReceipt<PNumber, PHash, Hash>,
```

Comment:
Yeah, it's odd indeed and should be the same type in theory, but for this usage https://github.com/subspace/subspace/blob/bf40b59a9a4b2e3ca803909a837bbde87b10c584/cumulus/client/cirrus-executor/src/bundle_processor.rs#L468 where `other` uses `NumberFor<PBlock>` while `local` uses `NumberFor<Block>`, so another generic `PNumber` here is just to make it compile for simplicity as it does not matter regarding the function :D

Reply:
Let's make a proper fix and make types the same if they should be. It is one of the worst things when types should be the same, but they are not and only work because they happen to be the same in the end. Types are part of documentation and we want documentation to be accurate.

Reply:
Interesting, now it just compiles :P